### PR TITLE
Handle "not in MPI" and "can't connect to MPI" errors differently

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Previous to this PR we were showing users a message saying that they were not in MPI if we couldn't even connect to MPI. With this PR we show different messages for these two scenarios:
- we cannot connect to MPI
- we could connect to MPI and the user is not in MPI

## Testing done
Local + unit tests + Cypress tests

## Screenshots
<details>
  <summary>cannot connect to MPI error</summary>
  
MPI error

</details>

<details>
  <summary>not in MPI error</summary>
  
image

</details>

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by erikphansen** [comment]
@tressaellen can you review the two error screenshots I attached to this PR description?

File: src/applications/personalization/profile-2/components/AccountSecurityContent.jsx [hunk 1]
```diff
@@ -69,7 +73,8 @@ export const AccountSecurityContent = ({
   return (
     <>
       {!isIdentityVerified && <IdentityNotVerified />}
-      {!isInMVI && isIdentityVerified && <NotInMVI />}
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/AccountSecurityContent.jsx:72]: the logic to determine if we show the `not in MPI` error got more complex with the inclusion of the new `can't connect to MPI` error; this is the main reason I moved the conditionals to `mapStateToProps`

File: src/applications/personalization/profile-2/tests/e2e/profile.mpi-error.cypress.spec.js [hunk 2]
```diff
@@ -29,7 +29,13 @@ function test(mobile = false) {
 
   // Should show an error alert about not being able to connect to MPI
   // TODO: We might show a different error message in this particular case since in this case we can't connect to MPI.
-  cy.findByText(/We can’t match your information to our Veteran records/i)
+  cy.findByText(/We can’t access your Veteran records/i)
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/tests/e2e/profile.mpi-error.cypress.spec.js:32]: the change in the component rendering logic means that this test now shows the "can't connect to MPI" error.

File: src/applications/personalization/profile-2/tests/e2e/profile.not-in-mpi-error.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import { PROFILE_PATHS } from '../../constants';
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/tests/e2e/profile.not-in-mpi-error.cypress.spec.js:1]: This test is nearly identical to the existing `mpi-error` test. The difference is we are using a different mock user and are checking for a different error message.

**Review by rianfowler — APPROVED** [review]

File: src/applications/hca/helpers.jsx [hunk 4]
```diff
@@ -16,14 +16,14 @@ import {
 } from 'platform/forms-system/src/js/helpers';
 import { getInactivePages } from 'platform/forms/helpers';
 import { isValidDate } from 'platform/forms/validations';
-import { isInMVI } from 'platform/user/selectors';
+import { isInMPI } from 'platform/user/selectors';
```
Comment (sanlouise) [inline src/applications/hca/helpers.jsx:19]: 👍

File: src/applications/personalization/profile-2/components/AccountSecurityContent.jsx [hunk 5]
```diff
@@ -6,7 +6,8 @@ import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
 import recordEvent from 'platform/monitoring/record-event';
 import {
   isLOA3 as isLOA3Selector,
-  isInMVI as isInMVISelector,
+  isInMPI as isInMVISelector,
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/components/AccountSecurityContent.jsx:9]: Can we do `isInMPISelector`?

File: src/applications/personalization/profile-2/components/AccountSecurityContent.jsx [hunk 6]
```diff
@@ -120,12 +125,21 @@ export const mapStateToProps = state => {
   const { verified, mhvAccount } = profile;
   const showMHVTermsAndConditions =
     verified && MHVTermsAndConditionsStatus.willRenderContent(mhvAccount);
+  const isInMVI = isInMVISelector(state);
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/components/AccountSecurityContent.jsx:128]: we should probably change MVI to MPI across the file

File: src/applications/personalization/profile-2/components/MPIConnectionError.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+
+const MPIConnectionError = () => {
+  const content = (
+    <>
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/components/MPIConnectionError.jsx:6]: Do we need the `<>` here?

File: src/applications/personalization/profile-2/tests/components/NotInMVI.unit.spec.jsx [hunk 8]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { mount } from 'enzyme';
 import { expect } from 'chai';
 
-import NotInMVI from '../../components/NotInMVI';
+import NotInMVI from '../../components/NotInMPIError';
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/NotInMVI.unit.spec.jsx:5]: Can we change this also to MPI?

**Review by sanlouise — APPROVED** [review]

**Comment by erikphansen** [comment]
Thanks for catching all my sloppy mistakes @sanlouise. Will fix!

**Pull request merged**